### ISSUE TITLE
Update on row detail are collapsing the expanded rows

### ISSUE THREAD

**Issue by vdumont54** [ghiscoding/Angular-Slickgrid#312]
### Your Environment

| Software          | Version(s) |
| ----------------- | ---------- |
| Angular           | 6.1       | 
| Angular-Slickgrid | 2.11.3        | 
| TypeScript        | 2.9.1        | 
| Operating System  | Windows 10        | 
| NPM/Node/Yarn     | 10.13.0        | 

### Context

In our project, we need to fetch data from our database which is update every seconds and we must see the update in real time.
Our grid use the row detail feature to show more data. 

### Expected Behavior

When an update occurs and if one or more rows are expanded, the rows must stay in this state.
![image](https://user-images.githubusercontent.com/33358810/66927695-3ca9bf80-effe-11e9-9c13-a9e2966c37d0.png)

### Current Behavior

When the update occurs, the expanded rows are not expanded or are collapsing..
![image](https://user-images.githubusercontent.com/33358810/66927653-269bff00-effe-11e9-92d7-4eadd60573a4.png)

### Code Sample

Here an [example](https://github.com/vdumont54/slickgrid-stream-example) of the current behavior.

**Comment by ghiscoding** [comment]
That's normal since SlickGrid does a re-render of the entire grid after a row changes, there's nothing I can do about that. What you could try is to keep track of which rows are expanded and re-apply them after a save.  The plugin has a few events that you can see [here](https://github.com/6pac/SlickGrid/blob/master/plugins/slick.rowdetailview.js#L779) you can maybe use `onAfterRowDetailToggle` and that would be `(sgOnAfterRowDetailToggle)` in Angular-Slickgrid and there's the method `getExpandedRows()` [here](https://github.com/6pac/SlickGrid/blob/master/plugins/slick.rowdetailview.js#L772) but to call a method directly from the SlickGrid plugin/addon, you'll have to follow this [example](https://github.com/ghiscoding/Angular-Slickgrid/wiki/Row-Detail#changing-addon-options-dynamically), you can then call `expandDetailView()` method (from the plugin again) by passing the row object, which is what `getExpandedRows()` returns I think.

It might be a long procedure but that is the best you can go, Angular-Slickgrid has nothing to do with this behavior since this plugin exist in SlickGrid (core lib) and Angular-Slickgrid only exposes it.

**Comment by ghiscoding** [comment]
again there isn't much that Angular-Slickgrid can do about this since it is a plugin from the core lib. I'm not surprised of the flashing since it will re-render without any row detail panel open after the update, so yeah it flashes because it renders without any panel and then it re-opens the same panel(s). SlickGrid render are required by any updates that would require to re-render DOM elements and an update of a row object is exactly that. 

Basically since Row Detail is an addon to the core code, the render method in the core has no knowledge of a Row Detail and every time there's a render, it will render as regular rows without any panel. I don't think there's any way to go around the flashing issue

**Comment by vdumont54** [comment]
Thank you for your help.
We will think to another way to do what we want.

**Comment by ghiscoding** [comment]
Actually there's maybe 1 more test you can do, I'm not sure if the `grid.render()` actually does close the Row Detail panel or if it's from `grid.invalidateRows()`. To test this out, try calling a `render` from a button just to see if it closes the Row Detail, if it doesn't then there might be a possibility to use the grid object methods. Where you using the GridService to do your update of the row?

On a button action, call this just to see if it closes the Row Detail
```ts
triggerRender() {
  this.angularGrid.slickGrid.render();
}
```

**Comment by vdumont54** [comment]
I try and `this.angularGrid.slickGrid.render()` didn't close the detail, but when i update the row using
`this.angularGrid.slickGrid.updateRow(row)` it is closing the detail.

**Comment by ghiscoding** [comment]
I think it's because of this [line](https://github.com/6pac/SlickGrid/blob/master/plugins/slick.rowdetailview.js#L167) in the plugin, it does call `invalidateRows`

**Comment by ghiscoding** [comment]
You can go to your `node_modules` folder and open the plugin and modify the line I just identified
`\node_modules\slickgrid\plugins\slick.rowdetailview.js` 

Change from  
```js
_handler.subscribe(_grid.getData().onRowsChanged, function (e, a) {
        _grid.invalidateRows(a.rows);
        _grid.render();
});
```
to this
```js
_handler.subscribe(_grid.getData().onRowsChanged, function (e, a) {
        _grid.updateRow(a.rows);
        _grid.render();
});
```
and that will probably fix your issue (you might need to clear your cache to get the JS code)

**Comment by vdumont54** [comment]
I have made the change and I get this result
![image](https://user-images.githubusercontent.com/33358810/66944657-9bc9fd00-f01b-11e9-9dde-4727a8dce8bb.png)

And when there is an update I get 
![image](https://user-images.githubusercontent.com/33358810/66944705-b56b4480-f01b-11e9-9850-a7fa0bffa361.png)

**Comment by ghiscoding** [comment]
hmm well that might be why it needs to use `invalidateRows`, which portion are you updating the Row Detail ("My Detail") or the row in the grid? I don't think that would work if you try to update the Row Detail.

**Comment by ghiscoding** [comment]
You can add a `console.log` in the `onRowsChanged` subscribe, I have a feeling that it's calling multiple rows not just that one and that is why it needs to call the `invalidateRows`. It's the side effect of being an external plugin, it's not built in the core itself. I again think you're stuck with current behavior, I'll let you play with it and if you find a solution you can comment
